Formula/mono.rb

```diff
@@ -38,16 +38,57 @@ class Mono < Formula
   link_overwrite "lib/cli"
 
   resource "fsharp" do
-    url "https://github.com/fsharp/fsharp.git",
-        tag:      "10.2.3",
-        revision: "e31bc96e8a5e5742af1c6c45d55d5cc06bb524cb"
+    url "https://github.com/dotnet/fsharp.git",
+        tag:      "v11.0.0-beta.20471.5",
+        revision: "03283e07f6bd5717797acb288cf6044cedca2202"
+    # F# patches hhen upgrading Mono, make sure to use the revision from
+    # https://github.com/mono/mono/blob/mono-#{version}/packaging/MacSDK/fsharp.py
+    patch do
+      url "https://raw.githubusercontent.com/mono/mono/a22ed3f094e18f1f82e1c6cead28d872d3c57e40/packaging/MacSDK/patches/fsharp-portable-pdb.patch"
+      sha256 "5b09b0c18b7815311680cc3ecd9bb30d92a307f3f2103a5b58b06bc3a0613ed4"
+    end
+    patch do
+      url "https://raw.githubusercontent.com/mono/mono/a22ed3f094e18f1f82e1c6cead28d872d3c57e40/packaging/MacSDK/patches/fsharp-netfx-multitarget.patch"
+      sha256 "112f885d4833effb442cf586492cdbd7401d6c2ba9d8078fe55e896cc82624d7"
+    end
+    patch do
+      url "https://github.com/dotnet/fsharp/commit/be6b22d11ae996b2d9b8e0724d9cf05ad65a0485.patch?full_index=1"
+      sha256 "793a39da798673b99289f3ac344ff8bd23d7eea2d3366c28e7e42229d8b130ca"
+    end
+  end
+
+  resource "fsharp-layout-patch" do
+    url "https://raw.githubusercontent.com/mono/mono/master/packaging/MacSDK/fsharp-layout.sh"
+    sha256 "f2cc63bf77e50663d91c6d102ba1d9217d1b9100c57071f79f0ae5a45e80ef42"
```

Comment:
```suggestion
    url "https://raw.githubusercontent.com/mono/mono/3070886a1c5e3e3026d1077e36e67bd5310e0faa/packaging/MacSDK/fsharp-layout.sh"
    sha256 "f2cc63bf77e50663d91c6d102ba1d9217d1b9100c57071f79f0ae5a45e80ef42"
```
Thanks for your work on this. Let's pin this to a specific commit hash since `master` is a moving target.

Reply:
Sure.